client/components/data/query-site-comments/index.jsx

```diff
@@ -9,7 +9,9 @@ import { connect } from 'react-redux';
  */
 import { requestCommentsList } from 'state/comments/actions';
 
-const requestComments = ( { requestSiteComments, siteId, status } ) => requestSiteComments( { siteId, status } );
+const requestComments = ( { requestSiteComments, siteId, status } ) => (
+	siteId && requestSiteComments( { siteId, status } )
```

Comment:
@Copons I also added a check here so we wouldn't query if siteId was falsey. You can get into this state if we have a clean indexedDB cache.